### PR TITLE
SaraSchema Version 2

### PR DESCRIPTION
**Pull request by Bialogs**
* Update base schema to draft-07 of json-schema.org
* Incorporates Twilio Errors into base schema through error_code schema
* Incorporates Execution IDs into patient_submission_token schema
* Incorporates recognized Twilio response status through response_status_schema
* Incorporates short and long-form patient submission tokens into patient_submission_token schema
* Allows all fields to be null
* Fixes various validation issues
* Expand test data structures with most common assessment types and edge cases
* Add missing properties to reported_symptoms_array schema

### CODE REVIEW

File: schemas/assessment.json [hunk 1]
```diff
@@ -5,30 +5,43 @@
...
     },
     "experiencing_symptoms": {
       "description": "Represents an assessment response over voice.",
-      "anyOf": [
-        { "type": "boolean" },
-        { "type": "null" }
-      ]
+      "type": ["boolean", "null"]
     },
     "reported_symptoms_array": {
```
Comment (ngfreiter) [inline schemas/assessment.json:37]: Can't comment on the actual line, but in the `reported_symptoms_array`, I think we need to add a property `required` to the list of required properties. So
```json
        "required": [
          "name",
          "value",
          "type",
          "label",
          "notes",
          "required"
        ],
```
Since from what I'm seeing, `required` is a boolean we set on the symptoms array [here](https://github.com/SaraAlert/SaraAlert/blob/9cdc2fe10ede7ee09f5701f779b57203b5468aeb/app/controllers/assessments_controller.rb#L135). So for me the validation was failing because this additional disallowed property was present.

**Review by ngfreiter — CHANGES_REQUESTED** [review]

File: schemas/assessment.json [hunk 2]
```diff
@@ -5,30 +5,43 @@
...
+      "description": "Unique token used to validate existence of a patient for this assessement's submission. Supports old 40 character token, new 10 character tokens, and 34 character execution flow IDs",
+      "type": ["string", "null"],
+      "anyOf": [
+        { "pattern": "^[a-f0-9]{40}$" },
+        { "pattern": "^[a-zA-Z0-9_-]{10}$" },
+        { "pattern": "^FN[a-f0-9]{32}$" }
+      ]
     },
     "experiencing_symptoms": {
       "description": "Represents an assessment response over voice.",
```
Comment (mayerm94) [inline schemas/assessment.json]: voice * or SMS

**Review by ngfreiter — APPROVED** [review]
Looks good to me!
